lib/HLSL/DxilModule.cpp

```diff
@@ -925,8 +925,8 @@ void DxilModule::LoadDxilSamplerFromMDNode(llvm::MDNode *MD, DxilSampler &S) {
 template <typename TResource>
 static void RemoveResources(std::vector<std::unique_ptr<TResource>> &vec,
                     std::unordered_set<unsigned> &immResID) {
-  for (std::vector<std::unique_ptr<TResource>>::iterator p = vec.begin(); p != vec.end();) {
-    std::vector<std::unique_ptr<TResource>>::iterator c = p++;
+  for (typename std::vector<std::unique_ptr<TResource>>::iterator p = vec.begin(); p != vec.end();) {
```

Comment:
I wonder whether we can use `auto` here.

Reply:
Yeah. that's actually what I did first, but I noticed that an earlier change removed auto in similar places . So I wasn't sure if that was something to be avoided since you need C++11 for auto. I'm happy to do it either way though.

Reply:
We use C++11 features in many places. I too think auto would be nicer.

Reply:
Okay then. Coming up!